chanzuckerberg/idseq-web
# Show search results stats in new sidebar

# Description

This adds a sidebar to the new library and public pages which shows stats on the current search results. 

![image](https://user-images.githubusercontent.com/28797/53525946-d0427600-3a97-11e9-96c6-83a8a8ec25e6.png)

The metadata values are clickable. The effect in future will be to filter the current results. (Question: will it add a filter to the current filter set or replace the current set?)

The sidebar shows for the projects view as well, but not visualizations, because it not clear how or why we would summarize their metadata.

![image](https://user-images.githubusercontent.com/28797/53526070-231c2d80-3a98-11e9-9a48-8d1861908981.png) 

The data for both views currently comes from the same api calls that the `DiscoveryView` makes. That means they only summarize what is currently loaded, not all possible results! (@tfrcarvalho and I have discussed other data approaches)

See @happyimadesignr mocks https://chanzuckerberg.invisionapp.com/share/EAQPA523SD9#/screens/348888814 . 

The main differences are that 

* Some of the stats are missing such as "non-host reads". It is not clear how to get these both efficiently for projects and samples, because they are not returned in the initial fetches. 
* There is no date graphic. This we can do in a later PR. (Note: I think treating date as a text field is still useful.)
* "Sample info" host and tissue are not shown. I'm not sure what these mean given that metadata also appears below. @happyimadesignr ?
* The "summary info / similar data" tabs do not exist. This will come in a later PR when similar data is ready. 
* I renamed "Sample info" to "Overall" to better reflect its contents. 
* I added a count of projects.
* The whole page width is different, pending addition of the left hand filters.
* I sorted the metadata values alphabetically. (Question: should this be by frequency?)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. go to /library
2. wait for samples
3. see sidebar filled in with data
4. click on project tab
5. see sidebar filled in
6. go to /public
7. repeat 2 thru 5 above
